adds external secrets bundle

chrisstafford opened the pull request:
It looks like nerc-prod was already pulling in the common configs. This PR just adds the config bundle and adds the externalsecrets directory under the ocp-prod overlay.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202467165231357

larsks commented on cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml:
```diff
@@ -5,4 +5,6 @@ commonLabels:
 
 resources:
 - ../common
+- ../../bundles/external-secrets-clustersecretstore
```
We do not actually want to set up a clustersecretstore on ocp-prod, because this would potentially expose secrets to all cluster users. We're going to want to set up namespaced secret stores where we need them.

larsks replied:
...so if this were to simply pull in `../../bundles/external-secrets` instead it would be fine, and then we would add the namespaced resources in a separate PR.

joachimweyl replied:
@chrisstafford can you take a look at this request?

chrisstafford replied:
I'll take care of this

joachimweyl replied:
@larsks did the implemented fix resolve your CR?

larsks committed:
Remove ClusterSecretStore

We don't want a ClusterSecretStore on nerc-ocp-prod because that could
inadvertently expose secrets to non-admin namespaces.

larsks committed:
Implement SecretStores as components

We are using namespaced SecretStores on ocp-prod in order to avoid
inadvertently exposing secrets through a ClusterSecretStore. These
resources are all going to be substantially identical, so we can implement
them using Kustomize components [1].

As part of this change:

- All namespaced SecretStores are named "nerc-secret-store"
- They all use a "vault-secret-reader" ServiceAccount
- They all point at the same vault instance

[1]: https://github.com/kubernetes-sigs/kustomize/blob/master/examples/components.md
